userguide/sound_classifier/how-it-works.md

```diff
@@ -0,0 +1,52 @@
+# How Does this Work?
+
+Training and making predctions, for a sound classifier model, is a three
+stage process. The first stage does signal preprocessing. In
+the second step we use a pretrained neural network to extract deep
+features. Finally in the last step, we use a custom neural network.
+
+## Signal Preprocessing Pipeline Stage
+Several audio transformation happen during this stage. For someone without
+sound domain knowledge, this is probably the most complicated part.
+Nothing about this stage is updated based on the input data (i.e. nothing
+is learned in this stage).
+
+At a high level the preprocessing pipeline does the following:
+* The raw pulse code modulation data from the wav file is converted to
+floats on a [-1.0, +1.0] scale.
+* The data is resampled to only 16,000 sample per second.
+* The data is broken up in to several overlapping windows.
+* A [Hamming Window](https://en.wikipedia.org/wiki/Window_function#Hann_and_Hamming_windows) is applied to each windows.
+* The [Power Spectrum](https://en.wikipedia.org/wiki/Spectral_density#Power_spectral_density) is calculated, using a [Fast Fourier Transformation](https://en.wikipedia.org/wiki/Fourier_transform).
+* Frequencies above and below certain thresholds are dropped.
+* Mel Frequency Filter Banks are applied.
```

Comment:
A Wikipedia link here would be nice: https://en.wikipedia.org/wiki/Mel-frequency_cepstrum

Reply:
Yeah, I debated about adding that link here. The thing is that link covers more than what's being mentioned in this bullet point. I agree this bullet point needs a link to more info.

I've added wikipedia links to both "Mel Frequency" and "Filter Banks". I think that more accurate and probably a lot easier to understand too.